File: 238_ProductOfArrayExceptSelf.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> sol (nums.size(), 1);
        vector<int> sample (nums.size());
        for (int i = 0; i < nums.size(); i++){
            sample[i] = nums[i];
        }
        int total = 1;
        for (int i = 0; i < nums.size(); i++){
            sol[i] *= total;
            total *= sample[i];
        }
        total = 1;
        for (int i = nums.size()-1; i > -1; i--){
            sol[i] *= total;
            total *= sample[i];
        }
        return sol;
    }
};

File: Leetcode_2023/238_ProductOfArrayExceptSelf.java
import java.util.*;
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] sol = new int[nums.length];
        sol[0] = 1;
        sol[1] = nums[0];
        for (int i = 2; i < nums.length; i++){
            sol[i] = sol[i-1]*nums[i-1];
        }
        int extra = nums[nums.length-1];
        sol[nums.length-2] *= extra;
        for (int i = nums.length-3; i > -1; i--){
            extra *= nums[i+1];
            sol[i] *= extra;
        }
        return sol;
    }
}